ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -496,6 +597,92 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
+        if adata.X is not None and var_names is not None:
+            idx = 0
+            for var_name in var_names:
+                if not var_name.startswith("ehrapycat"):
+                    break
+                idx += 1
+            # case: only encoded columns were found
+            if idx == len(var_names):
+                return None, None
+            # don't need to consider case when no encoded columns are there, since undo_encoding would not run anyways in this case
+            return adata.X[:, idx:].copy(), var_names[idx:]
```

Comment:
Copy might not be needed here, since this will be used in `hstack` later on, which will create another numpy array. Will check this out.